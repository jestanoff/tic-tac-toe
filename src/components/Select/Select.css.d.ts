// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'container': string;
  'fontPrimary': string;
  'select': string;
}
export const cssExports: CssExports;
export default cssExports;
