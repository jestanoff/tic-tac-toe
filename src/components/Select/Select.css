@value fontPrimary from '../../css/font.css';

.select {
    display: inline-block;
    font-family: fontPrimary;
    font-size: 12px;
    padding: 6px 20px;
}

.container {
    padding: 12px 12px 0;
    text-align: right;
}

.option {

}
