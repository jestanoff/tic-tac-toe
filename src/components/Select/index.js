import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styles from './Select.css';

const Select = ({ current, onChange, options }) => (
  <section className={styles.container}>
    <select onChange={onChange} value={current}>
      {options.map((option) => (
        <option value={option} key={option}>
          {option}
        </option>
      ))}
    </select>
  </section>
);

Select.propTypes = {
  current: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default memo(Select);
