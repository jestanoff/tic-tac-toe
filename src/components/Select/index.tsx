import * as React from 'react';
import styles from './Select.css';

type SelectProps = {
  current: string;
  handleChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
  options: string[];
};

const Select = ({ current, handleChange, options }: SelectProps) => (
  <section className={styles.container}>
    <select onChange={handleChange} title="Difficulty selector" value={current}>
      {options.map((option) => (
        <option value={option} key={option}>
          {option}
        </option>
      ))}
    </select>
  </section>
);

export default React.memo(Select);
