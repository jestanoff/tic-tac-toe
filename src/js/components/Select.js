import React, { PropTypes } from 'react';
import MenuItem from 'material-ui/MenuItem';
import SelectField from 'material-ui/SelectField';
import styles from '../../css/select.css';

const Select = ({ current, onChange, options }) => {
    const children = options.map((option, i) =>
        <MenuItem value={ option } key={ i } primaryText={ option } />,
    );

    return (
        <section className={ styles.container } >
            <SelectField
              value={ current } onChange={ onChange }
              selectedMenuItemStyle={ { color: '#1485bd' } }
            >
                { children }
            </SelectField>
        </section>
    );
};

Select.propTypes = {
    current: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    options: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Select;
