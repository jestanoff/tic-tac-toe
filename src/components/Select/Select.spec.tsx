import * as React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Select from './index';

describe('<Select />', () => {
  test('should render a select html control with options', () => {
    const { asFragment } = render(
      <Select current='Pick Me' options={['Choose Me', 'Pick Me']} handleChange={() => {}} />,
    );

    expect(screen.getByRole('combobox', { name: /Difficulty selector/i })).toBeInTheDocument();
    expect(screen.getAllByRole('option')).toHaveLength(2);
    expect(screen.getByRole('option', { name: /choose me/i })).toBeInTheDocument();
    expect(screen.getByRole('option', { name: /pick me/i })).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('should execute handleChange when a non active option is picked', async () => {
    expect.hasAssertions();
    const handleChangeMock = jest.fn();
    render(<Select current='1' options={['1', '2', '3']} handleChange={handleChangeMock} />);

    await userEvent.selectOptions(screen.getByRole('combobox', { name: /Difficulty selector/i }), '3')
    await waitFor(() => expect(handleChangeMock).toHaveBeenCalled());
    expect(handleChangeMock).toHaveBeenCalledTimes(1);
    expect(handleChangeMock).toHaveBeenCalledWith(expect.any(Object));
  });
});
