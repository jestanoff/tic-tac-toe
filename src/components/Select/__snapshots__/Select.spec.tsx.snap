// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render a select html control with options 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <select
      title="Difficulty selector"
    >
      <option
        value="Choose Me"
      >
        Choose Me
      </option>
      <option
        value="Pick Me"
      >
        Pick Me
      </option>
    </select>
  </section>
</DocumentFragment>
`;
